src/Controller/RecoveryController.php

```diff
@@ -0,0 +1,300 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\User;
+use App\Form\Model\RecoveryProcess;
+use App\Form\Model\RecoveryResetPassword;
+use App\Form\Model\RecoveryToken;
+use App\Form\Model\RecoveryTokenAck;
+use App\Form\RecoveryProcessType;
+use App\Form\RecoveryResetPasswordType;
+use App\Form\RecoveryTokenAckType;
+use App\Form\RecoveryTokenType;
+use App\Handler\RecoveryTokenHandler;
+use App\Helper\PasswordUpdater;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author doobry <doobry@systemli.org>
+ */
+class RecoveryController extends Controller
+{
+    /**
+     * @var PasswordUpdater
+     */
+    private $passwordUpdater;
+
+    /**
+     * @var RecoveryTokenHandler
+     */
+    private $recoveryTokenHandler;
+
+    /**
+     * RecoveryController constructor.
+     *
+     * @param PasswordUpdater      $passwordUpdater
+     * @param RecoveryTokenHandler $recoveryTokenHandler
+     */
+    public function __construct(PasswordUpdater $passwordUpdater, RecoveryTokenHandler $recoveryTokenHandler)
+    {
+        $this->passwordUpdater = $passwordUpdater;
+        $this->recoveryTokenHandler = $recoveryTokenHandler;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     */
+    public function recoveryProcessAction(Request $request): Response
+    {
+        $processState = 'NONE';
```

Comment:
This should be a constant.

Reply:
but `$processState` changes depending on the process state. Or do you mean constants for all states and using the relevant one then?

Reply:
Of course for all the States.

Reply:
I see. I changed them to constants now.